Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorInstance.cpp

```diff
@@ -815,6 +816,27 @@ namespace AZ::Render
         MeshComponentNotificationBus::Event(m_entityId, &MeshComponentNotificationBus::Events::OnObjectSrgCreated, objectSrg);
     }
 
+    void AtomActorInstance::UpdateLightingChannelMask()
+    {
+        if (m_meshHandle)
+        {
+            const AZStd::vector<Data::Instance<RPI::ShaderResourceGroup>>& objectSrgs =
+                m_meshFeatureProcessor->GetObjectSrgs(*m_meshHandle);
+            for (auto& objectSrg : objectSrgs)
+            {
+                RHI::ShaderInputConstantIndex lightingChannelMaskIndex =
+                    objectSrg->FindShaderInputConstantIndex(AZ::Name("m_lightingChannelMask"));
+                if (lightingChannelMaskIndex.IsValid())
+                {
+                    objectSrg->SetConstant(lightingChannelMaskIndex, m_actorInstance->GetLightingChannelMask());
+                    m_meshFeatureProcessor->QueueObjectSrgForCompile(*m_meshHandle);
```

Comment:
Do we need to queue the SRG compilation so many times in this method? Shouldn't one suffice at the end of the method?